options.go

```diff
@@ -450,3 +464,32 @@ func WithShrinkThread(enable bool, threshold int, delay time.Duration) Option {
 		return
 	})
 }
+
+// WithProfileReporter will enable reporter
+func WithProfileReporter(r ProfileReporter) Option {
+	return optionFunc(func(opts *options) (err error) {
+		if r == nil {
+			return nil
+		}
+
+		opts.pReportOpts.reporter = r
+		// active a background goroutine to consume profile report event.
+		if atomic.LoadInt32(&opts.pReportOpts.active) != 1 {
+
+			atomic.StoreInt32(&opts.pReportOpts.active, 1)
+			go func(cancel <-chan struct{}, evt <-chan func()) {
+				for {
+					select {
+					case <-cancel:
```

Comment:
we'd better always enable a background goroutine? we can omit this cancel channel and make code simpler then.